flink-runtime/src/test/java/org/apache/flink/runtime/jobmanager/DefaultJobGraphStoreTest.java

```diff
@@ -220,14 +221,23 @@ public void testGlobalCleanupWithNonExistName() throws Exception {
                 .globalCleanupAsync(testingJobGraph.getJobID(), Executors.directExecutor())
                 .join();
 
-        try {
-            removeFuture.get(timeout, TimeUnit.MILLISECONDS);
-            fail(
-                    "We should get an expected timeout because we are removing a non-existed job graph.");
-        } catch (TimeoutException ex) {
-            // expected
-        }
-        assertThat(removeFuture.isDone(), is(false));
+        removeFuture.join();
+        assertThat(removeFuture.isDone(), is(true));
```

Comment:
there's no way for this to fail.

Reply:
You're right. The `removeFuture.join()` is not necessary.